meson.build

```diff
@@ -0,0 +1,237 @@
+
+conf = configuration_data()
+deps = []
+
+# Compiler setup
+
+cc = meson.get_compiler('c')
+
+if cc.get_argument_syntax() == 'gcc'
+    cflags = [
+        '-Wno-sign-compare',
+        '-Wno-unused-parameter',
+        '-Werror-implicit-function-declaration',
+        '-Wstrict-prototypes',
+        '-Wpointer-arith',
+        '-Wredundant-decls',
+        '-Wno-missing-field-initializers'
+    ]
+    add_project_arguments(cc.get_supported_arguments(cflags), language: 'c')
+endif
+
+test_args = []
+
+if host_machine.system() == 'linux'
+    test_args += '-D_GNU_SOURCE'
+    add_project_arguments('-D_GNU_SOURCE', language: 'c')
+elif host_machine.system() == 'darwin'
+    test_args += '-D_DARWIN_C_SOURCE'
+    add_project_arguments('-D_DARWIN_C_SOURCE', language: 'c')
+elif cc.get_id() != 'msvc'
+    test_args += '-D_POSIX_C_SOURCE=200809L'
+    add_project_arguments('-D_POSIX_C_SOURCE=200809L', language: 'c')
+endif
+
+# Configuration
+
+if get_option('large-tiles')
+    conf.set('CONFIG_LARGE_TILES', 1)
+endif
+
+conf.set('PACKAGE_NAME', 'libass')
+conf.set('PACKAGE_VERSION', meson.project_version())
+
+conf.set('CONFIG_SOURCEVERSION', '"meson, on or after: ' + meson.project_version() + '"')
+
+str_check_functions = [
+    'strdup',
+    'strndup'
+]
+
+foreach name: str_check_functions
+    if cc.has_function(name) and cc.has_header_symbol('string.h', name, args: test_args)
+        conf.set('HAVE_@0@'.format(name.to_upper()), 1)
+    endif
+endforeach
+
+# Dependencies
+
+deps += cc.find_library('m', required: false)
+
+# Replace with Meson iconv lookup (`dependency('iconv')`) once 0.60 can be required, see: https://github.com/libass/libass/pull/330#discussion_r707666730
+iconv_dep = cc.find_library('iconv', required: false)
+if iconv_dep.found()
+    deps += iconv_dep
+endif
+if iconv_dep.found() or cc.has_function('iconv_open')
+    conf.set('CONFIG_ICONV', 1)
+endif
+
+freetype_dep = dependency('freetype2',version: '>= 2.10.92',
+                          fallback: ['freetype2', 'freetype_dep'],
+                          default_options: ['default_library=' + get_option('default_library'), 'harfbuzz=disabled']) # workaround for https://github.com/mesonbuild/meson/issues/8047
+if freetype_dep.found()
+    deps += freetype_dep
+    conf.set('CONFIG_FREETYPE', 1)
+endif
+
+if host_machine.system() == 'windows' and get_option('default_library') == 'static'
+    add_project_arguments('-DFRIBIDI_LIB_STATIC', language: 'c')
+endif
+fribidi_dep = dependency('fribidi', version: '>= 0.19.1',
+                         fallback: ['fribidi', 'libfribidi_dep'],
+                         default_options: ['docs=false', 'tests=false'])
+if fribidi_dep.found()
+    deps += fribidi_dep
+    conf.set('CONFIG_FRIBIDI', 1)
+endif
+
+harfbuzz_options = ['tests=disabled', 'cairo=disabled', 'gobject=disabled']
+if host_machine.system() == 'windows' or host_machine.system() == 'darwin'
+    harfbuzz_options += ['glib=disabled', 'freetype=disabled']
+endif
```

Comment:
Also applies to other dpendencies with custom options:
Why do we need to set options for our dependencies and why are they platform specific? Can they be removed?

Reply:
> And why is freetype disabled eventhough freetype will be available?

Not sure how this code originally came to be, but for modern libass it’s useless, for what it’s worth, and disabling it will save some space if HarfBuzz’s build script autodetect-enables it by default.

Reply:
> Why are they platform specific?

It's been a while, but my recollection was that glib didn't build nicely on Windows/Mac and I confirmed that it doesn't do anything. I'm not sure about Linux, but it probably could be excluded there too? It's just trivially available.

> And why is freetype disabled eventhough freetype will be available?

I think to break the circular dependency I disabled both. I'm happy to enable in the preferred order, which I guess is freetype-less harfbuzz into harfbuzz-enabled freetype?

Reply:
Right, we’re not doing anything with Harfbuzz’s FreeType support (anymore). Personally I wouldn't have added any default options, but if they are platform agnostic, stable and not too many I have no objections.

> None of HarfBuzz’s optional dependencies are useful for us except Graphite2

Side note: I may misremember, but I have a faint memory of some feature currently only being supported via CoreText(?). Even if that’s correct though, I don't remember if it was anything important for libass’ use case.

Reply:
Or rather: No objection if this won't conflict with options a sup-build may have defined for its dependency on the same lib (i.e. If libC uses meson, dependens on libass and e.g. harfbuzz and has set or chose to not set any options for its dependency on meson).

Reply:
> I have a faint memory of some feature currently only being supported via CoreText(?)

You’re probably thinking of AAT (Apple Advanced Typography). Funnily enough, now that you mention it, I also have such a faint memory; but looking at HarfBuzz’s current docs and Git history, I see no mention of Core Text ever having been required for it. At any rate, current HarfBuzz claims to support AAT natively, and I’m guessing on all platforms.